logic/src/androidMain/kotlin/com/wire/kalium/logic/CoreLogic.kt

```diff
@@ -37,10 +40,17 @@ actual class CoreLogic(
 
             val workScheduler = WorkScheduler(applicationContext, session)
             val syncManager = SyncManagerImpl(workScheduler)
-            AuthenticatedDataSourceSet(networkContainer, proteusClient, workScheduler, syncManager).also {
+            val encryptedSettingsHolder = EncryptedSettingsHolder(applicationContext, "${PREFERENCE_FILE_PREFIX}-${session.userId}")
+            val userPreferencesSettings = KaliumPreferencesSettings(encryptedSettingsHolder.encryptedSettings)
+            val database = Database(applicationContext, "main.db", userPreferencesSettings)
+            AuthenticatedDataSourceSet(networkContainer, proteusClient, workScheduler, syncManager, database, userPreferencesSettings, encryptedSettingsHolder).also {
```

Comment:
How about re-formatting like this since it's so long now.

```suggestion
            AuthenticatedDataSourceSet(networkContainer, 
                                       proteusClient, 
                                       workScheduler, 
                                        syncManager, 
                                        database, 
                                        userPreferencesSettings, 
                                        encryptedSettingsHolder).also {
```

Reply:
👍